usd/src/sdf_parser/Link.cc

```diff
@@ -0,0 +1,138 @@
+ *
+*/
+
+#include "sdf/usd/sdf_parser/Link.hh"
+
+#include <string>
+
+#include <ignition/math/Pose3.hh>
+
+// TODO(ahcorde) this is to remove deprecated "warnings" in usd, these warnings
+// are reported using #pragma message so normal diagnostic flags cannot remove
+// them. This workaround requires this block to be used whenever usd is
+// included.
+#pragma push_macro ("__DEPRECATED")
+#undef __DEPRECATED
+#include <pxr/usd/sdf/path.h>
+#include <pxr/usd/usd/stage.h>
+#include <pxr/usd/usdGeom/xform.h>
+#include <pxr/usd/usdPhysics/massAPI.h>
+#include <pxr/usd/usdPhysics/rigidBodyAPI.h>
+#pragma pop_macro ("__DEPRECATED")
+
+#include "sdf/Link.hh"
+#include "sdf/usd/sdf_parser/Light.hh"
+#include "../UsdUtils.hh"
+
+namespace sdf
+{
+// Inline bracke to help doxygen filtering.
+inline namespace SDF_VERSION_NAMESPACE {
+//
+namespace usd
+{
+  UsdErrors ParseSdfLink(const sdf::Link &_link, pxr::UsdStageRefPtr &_stage,
+      const std::string &_path, bool _rigidBody)
+  {
+    const pxr::SdfPath sdfLinkPath(_path);
+    UsdErrors errors;
+
+    auto usdLinkXform = pxr::UsdGeomXform::Define(_stage, sdfLinkPath);
+
+    ignition::math::Pose3d pose;
+    auto poseErrors = sdf::usd::PoseWrtParent(_link, pose);
+    if (!poseErrors.empty())
+    {
+      for (const auto &e : poseErrors)
+        errors.push_back(e);
+      return errors;
+    }
+
+    poseErrors = usd::SetPose(pose, _stage, sdfLinkPath);
+    if (!poseErrors.empty())
+    {
+      for (const auto &e : poseErrors)
+        errors.push_back(e);
+      errors.push_back(UsdError(UsdErrorCode::SDF_TO_USD_PARSING_ERROR,
+            "Unable to set the pose of the link prim corresponding to the "
+            "SDF link named [" + _link.Name() + "]"));
+      return errors;
+    }
+
+    if (_rigidBody)
+    {
+      auto linkPrim = _stage->GetPrimAtPath(sdfLinkPath);
+      if (!linkPrim)
+      {
+        errors.push_back(UsdError(sdf::usd::UsdErrorCode::INVALID_PRIM_PATH,
+              "Internal error: unable to get prim at path ["
+              + _path + "], but a link prim should exist at this path"));
+        return errors;
+      }
+
+      if (!pxr::UsdPhysicsRigidBodyAPI::Apply(linkPrim))
+      {
+        errors.push_back(UsdError(sdf::usd::UsdErrorCode::FAILED_PRIM_API_APPLY,
+              "Internal error: unable to mark link at path [" + _path
+              + "] as a rigid body, so mass properties won't be attached"));
+        return errors;
+      }
+
+      auto massAPI =
+        pxr::UsdPhysicsMassAPI::Apply(linkPrim);
+      if (!massAPI)
+      {
+        errors.push_back(UsdError(sdf::usd::UsdErrorCode::FAILED_PRIM_API_APPLY,
+              "Unable to attach mass properties to link ["
+              + _link.Name() + "]"));
+        return errors;
+      }
+      massAPI.CreateMassAttr().Set(
+          static_cast<float>(_link.Inertial().MassMatrix().Mass()));
+
+      const auto diagonalInertial =
+        _link.Inertial().MassMatrix().DiagonalMoments();
```

Comment:
I think it would be more proper to call [PrincipalMoments](https://ignitionrobotics.org/api/math/6.10/classignition_1_1math_1_1MassMatrix3.html#ade8f30327a251a55befa8a7495a9b323) here instead of `DiagonalMoments` and then to also call `massAPI.CreatePrincipalAxesAttr` and populate that with a quaternion from [PrincipalAxesOffset](https://ignitionrobotics.org/api/math/6.10/classignition_1_1math_1_1MassMatrix3.html#a2f5873ac88cab23f7391e6c34eb79730). If you don't want to do that now, please add a `TODO` comment and/or create an issue.